### ISSUE TITLE
thumbnails not reloaded after login

### ISSUE THREAD

**Issue by nicolasfranck** [UniversalViewer/universalviewer#929]
**UV version:**

```
 universalviewer@4.0.22
```

**I'm submitting a:**
- [x] bug report => please fork one of these [codesandbox examples](https://github.com/UniversalViewer/universalviewer/wiki/UV-Examples) with a repro of your issue and include a link to it below
- [ ] feature request => please use the [user stories repo](https://github.com/UniversalViewer/user-stories)
- [ ] support request => Please do not submit support requests here, use [stackoverflow](http://stackoverflow.com/questions/tagged/iiif)

**Current behavior:**

After login (in a separate tab), the viewer is reloaded, and is working,
but the thumbnails are not reloaded, and therefore look broken.

**Comment by nicolasfranck** [comment]
What I noticed:

* the content-left-panel uses a different ThumbsView: https://github.com/UniversalViewer/universalviewer/blob/main/src/content-handlers/iiif/modules/uv-contentleftpanel-module/ThumbsView.tsx, one that is built with React? That one is different from https://github.com/UniversalViewer/universalviewer/blob/main/src/content-handlers/iiif/modules/uv-shared-module/ThumbsView.ts where cache invalidation is configurable (by adding a timestamp). 
* When I add the same timestamp logic to the `thumb.uri` [here](https://github.com/UniversalViewer/universalviewer/blob/main/src/content-handlers/iiif/modules/uv-contentleftpanel-module/ThumbsView.tsx#L50) in that react component, the thumbnails are reloaded after login. But the method `renderThumbs` in content-left-panel is called a lot, so I do not know if this such a good idea.

**Comment by demiankatz** [comment]
@nicolasfranck, I like the idea of only changing the timestamp when needed... are you thinking that we have a `lastThumbnailCacheTime` value, and an event that resets it? And then we could trigger that event as part of the authentication workflow? Unfortunately, I don't know enough about the history or context of this part of the code to comment on the possible implications of one approach or another, though.

**Comment by nicolasfranck** [comment]
Yes, a cache key that is reset after certain events, like login. I think that in UV3 the date stamps were added as a common way to prevent stale cached requests (thumbnails that failed the first time due to missing cookies). But in UV4 the thumbnail tab is being rebuild more times than it was in UV3. Such a way would also make that configuration option for cache invalidation obsolete, not?

**Comment by demiankatz** [comment]
The cache invalidation option might still be useful even if we add a "smart cache key," if we define it as "always invalidate cache." This way, it may be useful as a debugging tool -- if thumbnails aren't displaying correctly and turning on the setting fixes the problem, then we know that our smart caching code must be missing an edge case somewhere. However, this moves it from being a general-purpose tool to being more of a debugging tool, so it might make sense to deprecate the old config name and create a new one for this purpose, if it's worth retaining it at all. As with all of my comments, though, I have limited insight into the existing code, so please take my thoughts with a grain of salt. ;-)
